### PR TITLE
Add unit tests for PetOperations

### PR DESCRIPTION
**Pull request by v-jiaodi**

### CODE REVIEW

File: petstore/clients/javascript/samples/javascript/petOperations.js [hunk 1]
```diff
@@ -34,8 +48,45 @@ async function main() {
...
+
+  // update a non existent pet, return error message is Pet not found
+  try {
+    await client.pets.update(15, {
+      name: "Test",
+      age: 5,
+      ownerId: 5,
+    });
+  } catch (e) {
+    console.log(e.message);
```
Comment (MaryGao) [inline petstore/clients/javascript/samples/javascript/petOperations.js:70]: assert pls

File: petstore/clients/javascript/test/sample.spec.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { PetStoreClient } from "@unbranded/petstore";
```
Comment (MaryGao) [inline petstore/clients/javascript/test/sample.spec.ts]: filename should be `petOperations.spec.ts`.

File: petstore/clients/javascript/test/sample.spec.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import { assert, beforeEach, it, describe } from "vitest";
+
+describe("Test Samples", () => {
```
Comment (MaryGao) [inline petstore/clients/javascript/test/sample.spec.ts]:
```suggestion
describe("PetOperations Samples", () => {
```

File: petstore/clients/javascript/test/sample.spec.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+        // update a invalid pet, return error message is Invalid petId
+        try{
+            await client.pets.update(-1,pet);
+        } catch (error) {
+            assert.equal(error.code, 0);
+            assert.equal(error.message, 'Invalid petId');
+        }
+
+        // update a non existent pet, return error message is Pet not found
```
Comment (MaryGao) [inline petstore/clients/javascript/test/sample.spec.ts]: split these test cases

File: petstore/clients/javascript/test/sample.spec.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        const result = await client.pets.get(1);
+        assert.equal(result.name, 'Kiwi');
+        assert.equal(result.age, 5);
+        assert.equal(result.ownerId, 5);
+
+        // get a invalid pet, return error message is Invalid petId
+        try{
+            await client.pets.get(-1);
+        } catch (error) {
+            assert.equal(error.code, 0);
```
Comment (MaryGao) [inline petstore/clients/javascript/test/sample.spec.ts]: really, this code is 0?

Comment (MaryGao) [reply petstore/clients/javascript/test/sample.spec.ts]: pls share the link here for code and message readme

Comment (v-jiaodi) [reply petstore/clients/javascript/test/sample.spec.ts]: yes. https://github.com/allenjzhang/typespec-e2e-demo/blob/main/petstore/servers/aspnet/implemented_apis.md

File: petstore/clients/javascript/test/petOperations.spec.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+  it("pet create", async function () {
+    const result = await client.pets.create(pet);
+    assert.equal(result.name, "Kiwi");
+    assert.equal(result.age, 5);
+    assert.equal(result.ownerId, 5);
+  });
+
+  it("pet update", async function () {
+    // update a pet, return a pet
```
Comment (MaryGao) [inline petstore/clients/javascript/test/petOperations.spec.ts:67]: split test cases also

**Review by MaryGao — APPROVED** [review]

File: petstore/clients/javascript/package.json [hunk 7]
```diff
@@ -40,7 +40,8 @@
     "pack": "npm pack 2>&1",
     "lint": "eslint package.json api-extractor.json src",
     "lint:fix": "eslint package.json api-extractor.json src --fix --fix-type [problem,suggestion]",
-    "build": "npm run clean && tshy && npm run extract-api"
+    "build": "npm run clean && tshy && npm run extract-api",
+    "test": "npx vitest -c vitest.config.ts"
```
Comment (MaryGao) [inline petstore/clients/javascript/package.json]: could you add a script to start test server?

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**